src/main/java/seedu/address/model/history/HistoryCommandList.java

```diff
@@ -0,0 +1,60 @@
+package seedu.address.model.history;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Comparator;
+import java.util.Iterator;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.logic.commands.AddCommand;
+import seedu.address.logic.commands.ClearCommand;
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.commands.DeleteCommand;
+import seedu.address.logic.commands.edit.EditCommand;
+
+/**
+ * A list of history commands.
+ */
+public class HistoryCommandList implements Iterable<HistoryCommand> {
+    private final ObservableList<HistoryCommand> internalList = FXCollections.observableArrayList();
+    private final ObservableList<HistoryCommand> internalUnmodifiableList =
+            FXCollections.unmodifiableObservableList(internalList);
+    private static String originalCommandText;
+
+    public HistoryCommandList() {
+    }
+
+    /**
+     * Adds a history command to the list.
+     * The person must not already exist in the list.
+     */
+    public void add(Command toAdd) {
+        requireNonNull(toAdd);
+        assert originalCommandText != null : "The original command text should not be null";
+        assert toAdd instanceof AddCommand
+                || toAdd instanceof ClearCommand
+                || toAdd instanceof DeleteCommand
+                || toAdd instanceof EditCommand : "History command is not changing the person list";
+
+        internalList.add(HistoryCommand.of(toAdd, originalCommandText));
```

Comment:
```suggestion
        internalList.add(0, HistoryCommand.of(toAdd, originalCommandText));
```

this is the fix for your current requirements

so you have a `getIndex` which returns index based on the order the commands were executed (which is great!)

so this means it is okay to store the elements in reverse, which is exactly what you need